enhancements/network/user-space-pod-iface-and-api-library.md

```diff
@@ -0,0 +1,262 @@
+  Unix socket file and virtio mode.
+  - Not needed by OCP immediately, but must be considered during the overall
+  design.
+
+This enhancement has two components:
+- How the data should be passed to the pod?
+- How to ease the consumption of the data in the pod?
+
+## Motivation
+
+Various system resources are allocated to run high performance network
+applications inside a container such as CPU cores, dedicated memory (hugepages),
+network devices, etc. However, these resources are presented differently inside
+the container and some of them are not visible to applications without
+explicitly injecting to the container.
+
+### Goals
+
+The goal is for network resource data information to be consistently passed
+to the container and provide an API library to facilitate the consumption of
+user space interfaces for applications (mainly DPDK) in containers, with the
+initial primary purpose of supporting SR-IOV interfaces.
+
+### Non-Goals
+
+The following items are out of scope for this proposal:
+- This could, secondarily, also be used to support user space data plane
+technologies like ovs-dpdk, but its not the same, and its not our primary goal.
+- Non-SR-IOV devices (vDPA and vhost-user interfaces) require additional
+information. However, these devices are not required for this enhancement.
+That being said, some of the upstream work this enhancement is pulling from does
+include those interfaces. So the API Library may include some of the additional
+work for these interfaces.
+
+## Proposal
+
+As described above, there are two parts to this enhancement. Work is need to
+pass additional data to the pod to enable user space applications to run
+properly. Work is also needed in the pod to ease the consumption of the data.
+Each item will be described below.
+
+### Passing Data to the Pod
+
+Currently, data is passed to the pod in several forms:
+- **PCI Addresses**: SR-IOV Device Plugin passes the list of PCI Addresses in
+the an environmental variable to the pod.
+- **Interface Data**: Multus populates the pod annotation with a
+'*k8s.v1.cni.cncf.io/networks-status*' field. This annotation contains the list
+of interfaces, and for each interface, the MAC address, list of IP addresses
+(not CIDR), interface (net1, net2, etc), name, etc. If the Kubernetes Downward
+API is enabled, this information is available to the pod.
+- **CPU List**: The dedicated CPU list can be retrieved by reading the
+'*cpuset*' from the cGroup in the pod.
+
+For user space applications running in pod, the following information is still
+needed:
+- **Hugepage Information**: The type and number of hugepages allocated to the
+pod.
+- **Additional Interface Data**: The '*k8s.v1.cni.cncf.io/networks-status*'
+contains a '*name*', which is the network in the CRD, but doesn't have enough
+information to determine which PCI Adress is associated with which interface.
+So addition information is needed o make that association within the pod.
```

Comment:
```suggestion
information to determine which PCI Address is associated with which interface.
So additional information is needed to make that association within the pod.
```

Also, not clear what the reference to `name` in the `network-status` is about; that's the purely-kubernetes-internal name of the network attachment. Did you mean `interface`?

Reply:
Added a footnote.